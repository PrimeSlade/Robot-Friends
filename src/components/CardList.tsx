import React from "react";
import Card from "./Card";
import { Robot } from "../type/Types";

type CardListProps = {
  robots: Array<Robot>;
};

const CardList = ({ robots }: CardListProps) => {
  return robots.map((user, i) => {
    return <Card key={i} id={user.id} name={user.name} email={user.email} />;
  });
};

export default CardList;
