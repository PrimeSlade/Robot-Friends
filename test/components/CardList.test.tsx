import { render, screen } from "@testing-library/react";
import CardList from "../../src/components/CardList";
import React from "react";
import { mockRobots } from "../__mocks__/mockRobots";

test("testing CardList by using snapshot", () => {
  // Mock robot data for testing
  const { asFragment } = render(<CardList robots={mockRobots} />);
  expect(asFragment()).toMatchSnapshot();
});

describe("CardList rendering logic", () => {
  beforeEach(() => {
    render(<CardList robots={mockRobots} />);
  });

  test("testing cards", () => {
    expect(screen.getAllByTestId("card")).toHaveLength(mockRobots.length);
  });

  test("testing rendered names", () => {
    mockRobots.forEach((r) => {
      expect(screen.getByText(r.name)).toBeInTheDocument();
    });
  });
});
