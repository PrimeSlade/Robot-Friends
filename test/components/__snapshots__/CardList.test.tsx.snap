// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CardList by using snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    data-testid="card"
  >
    <img
      alt="robot"
      src="https://robohash.org/1?200*200"
    />
    <div>
      <h2>
        Robot One
      </h2>
      <p>
        one@robots.com
      </p>
    </div>
  </div>
  <div
    class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    data-testid="card"
  >
    <img
      alt="robot"
      src="https://robohash.org/2?200*200"
    />
    <div>
      <h2>
        Robot Two
      </h2>
      <p>
        two@robots.com
      </p>
    </div>
  </div>
  <div
    class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    data-testid="card"
  >
    <img
      alt="robot"
      src="https://robohash.org/3?200*200"
    />
    <div>
      <h2>
        Robot Three
      </h2>
      <p>
        three@robots.com
      </p>
    </div>
  </div>
</DocumentFragment>
`;
